plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -247,7 +247,14 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
             throw new UnknownTableTypeException(tableName);
         }
 
-        org.apache.iceberg.Table table = getIcebergTable(session, hiveTable.get().getSchemaTableName());
+        TableMetadata metadata = tableMetadataCache.computeIfAbsent(
```

Comment:
Why not change `getIcerbergTable` itself if the goal is to prevent working with v2 tables?

Reply:
Good point. I'm not sure why I had it set in my head that this needed to go here. Thanks